WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/AuthorSelectionAdapter.kt

```diff
@@ -0,0 +1,137 @@
+    }
+
+    override fun getCount(): Int = items.count()
+
+    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
+        var view: View? = convertView
+        val holder: DropdownViewHolder
+
+        if (view == null) {
+            val inflater = LayoutInflater.from(parent.context)
+            view = inflater.inflate(R.layout.post_list_author_selection_dropdown, parent, false)
+            holder = DropdownViewHolder(view)
+            view.tag = holder
+        } else {
+            holder = view.tag as DropdownViewHolder
+        }
+
+        holder.bind(items[position], imageManager, uiHelpers)
+
+        return view!!
+    }
+
+    override fun getItem(position: Int): AuthorFilterListItemUIState = items[position]
+
+    override fun getItemId(position: Int): Long = position.toLong()
+
+    override fun getItemViewType(position: Int): Int = 0
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+        var view: View? = convertView
+        val holder: NormalViewHolder
+
+        if (view == null) {
+            val inflater = LayoutInflater.from(parent.context)
+            view = inflater.inflate(R.layout.post_list_author_selection, parent, false)
+            holder = NormalViewHolder(view)
+            view.tag = holder
+        } else {
+            holder = view.tag as NormalViewHolder
+        }
+
+        holder.bind(items[position], imageManager, uiHelpers)
+
+        return view!!
+    }
+
+    override fun getViewTypeCount(): Int = 1
+    override fun hasStableIds(): Boolean = false
+    override fun isEmpty(): Boolean = false
+
+    fun notifyDataSetChanged() {
+        for (observer in observers) {
+            observer.get()?.onChanged()
+        }
+    }
+
+    override fun registerDataSetObserver(observer: DataSetObserver) {
+        for (alreadyAdded in observers) {
+            if (observer == alreadyAdded.get()) {
+                // No need to add since it is already added
+                return
+            }
+        }
+
+        observers.add(WeakReference(observer))
+    }
+
+    fun updateItems(newItems: List<AuthorFilterListItemUIState>) {
+        items.clear()
+        items.addAll(newItems)
+        notifyDataSetChanged()
+    }
+
+    override fun unregisterDataSetObserver(observer: DataSetObserver) {
+        for ((index, currentRef) in observers.withIndex()) {
+            if (observer == currentRef.get()) {
+                observers.removeAt(index)
+                return
+            }
+        }
+    }
+
+    private open class NormalViewHolder(protected val itemView: View) {
+        protected val image: AppCompatImageView = itemView.findViewById(R.id.post_list_author_selection_image)
+
+        @CallSuper
+        open fun bind(state: AuthorFilterListItemUIState, imageManager: ImageManager, uiHelpers: UiHelpers) {
+            val avatarSize = image.resources.getDimensionPixelSize(R.dimen.avatar_sz_small)
```

Comment:
Can we get the dimension from `AuthorFilterListItemUIState` instead?

Reply:
We could, however give that it is the same no matter the UI state, and intrinsically linked to the layout files, I don't think it really adds anything to include that static data.

I would prefer to perhaps put that as a constant in this file so it is easier to be changed when the layout is updated if that would be a good compromise?

Reply:
I don't think having this as constant is necessary since it's a dimension. While @malinajirka and I were discussing this we wanted it to be consistent with the `AuthorFilterListItemUIState.dropDownBackground` so all the initiation happens in one place. I think we were both on the edge on this and since we don't have any plans to dynamically update the size, if you feel it's better to keep here let's go with that.

Reply:
I prefer it to be where it is now unless it is actually going to be dynamic.

Reply:
I agree it's fine to keep it here. 

SideNote: In an ideal world, both `getDimensionPixelSize` and `fixGravatarUrl` would be injectable/mockable methods and the ViewHolder would receive just the url, but we are not there yet.

Reply:
@malinajirka I could add the `getDimensionPixelSize` call into `UiHelpers` which is currently being injected. Get us one step closer?

Reply:
Thanks for the suggestion! I wouldn't do it as part of this PR as it's just a minor issue. Honestly, I'm not even sure `UiHelpers` would be the best place. We are currently using it just in the view layer and it has Android dependencies in parameters. But it's for another discussion :P.